### PR TITLE
Midgard API health check ready to rumble

### PR DESCRIPTION
**Pull request by mr-rooftop**

### CODE REVIEW

File: thornode_bot.py [hunk 1]
```diff
@@ -369,6 +416,17 @@ def get_status_endpoint():
...
 
+def get_midgard_endpoint():
+    """
+        Return the endpoint for Midgard API check
+    """
+
+    if DEBUG:
+        return 'http://localhost:8000/midgard.json'
+
+    return 'http://' + NODE_IP + ':8080/v1/health'
```
Comment (rekt0x) [inline thornode_bot.py:427]: Could be a single line of code:
```
return 'http://localhost:8000/midgard.json' if DEBUG else 'http://' + NODE_IP + ':8080/v1/health'
```

File: thornode_bot.py [hunk 2]
```diff
@@ -285,6 +292,31 @@ def check_block_height(context):
...
+    """
+        Check that Midgard API is ok
+    """
+    chat_id = context.job.context['chat_id']
+    user_data = context.job.context['user_data']
+
+    if 'is_midgard_healthy' not in user_data:
+        user_data['is_midgard_healthy'] = True
+
+    if user_data['is_midgard_healthy'] and not is_midgard_healthy():
```
Comment (rekt0x) [inline thornode_bot.py:305]: Maybe write out the if check, so it's easier readable:
```
... and is_midgard_healthy() == False:
```

File: thornode_bot.py [hunk 3]
```diff
@@ -285,6 +292,31 @@ def check_block_height(context):
...
+        user_data['is_midgard_healthy'] = False
+        text = 'Midgard API is not healthy anymore! 💀' + '\n' + \
+               'IP: ' + NODE_IP + '\n\n' + \
+               'Please check your Thornode immediately!'
+        context.bot.send_message(chat_id, text)
+        show_action_buttons(context, chat_id=chat_id)
+    elif not user_data['is_midgard_healthy'] and is_midgard_healthy():
```
Comment (rekt0x) [inline thornode_bot.py:312]: Maybe write out the if check, so it's easier readable:
```
elif user_data['is_midgard_healthy'] == True and ...
```

Comment (mr-rooftop) [reply thornode_bot.py:312]: And I was sure you would say "rewrite user_data['is_midgard_healthy'] == True to
user_data['is_midgard_healthy']"

**Pull request merged**